hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -121,11 +121,12 @@ public final Data toData(Object obj, PartitioningStrategy strategy) {
         BufferObjectDataOutput out = pool.takeOutputBuffer();
         try {
             SerializerAdapter serializer = serializerFor(obj);
-            out.writeInt(serializer.getTypeId(), ByteOrder.BIG_ENDIAN);
-            serializer.write(out, obj);
-
             int partitionHash = calculatePartitionHash(obj, strategy);
             out.writeInt(partitionHash, ByteOrder.BIG_ENDIAN);
+
```

Comment:
I\m fine with changing the order. But I'm curious about the reasoning. Personally I like it in the sense that it becomes very easy to gather all the meta data like partition id since the payload is at the end.